src/lib/index.ts

```diff
@@ -38,8 +38,15 @@ class TPAStylePlugin {
     );
     this.replaceRuntimeModule(compiler);
 
+    const pluginDescriptor = isWebpack5
+      ? TPAStylePlugin.pluginName
+      : {
+          name: TPAStylePlugin.pluginName,
+          stage: compilation.PROCESS_ASSETS_STAGE_OPTIMIZE,
+        };
+
     compiler.hooks.compilation.tap(TPAStylePlugin.pluginName, compilation => {
-      compilation.hooks.optimizeChunkAssets.tapAsync(TPAStylePlugin.pluginName, (chunks, callback) => {
+      compilation.hooks.processAssets.tapAsync(pluginDescriptor, (chunks, callback) => {
```

Comment:
`optimizeChunkAssets` has been deprecated and will be removed in v6. This keeps the same functionality on v5 and above.